lib/real-device.js

```diff
@@ -106,22 +109,27 @@ export class RealDevice {
       await installWithIosDeploy();
     } else {
       const afcService = await services.startAfcService(this.udid);
+      const enableParallelPush = _.toLower(/** @type {'parallel'} */ (strategy)) === APP_INSTALL_STRATEGY.PARALLEL;
       try {
         const bundleId = await extractBundleId(app);
         const bundlePathOnPhone = path.join(INSTALLATION_STAGING_DIR, bundleId);
-        await pushFolder(afcService, app, bundlePathOnPhone, {
-          timeoutMs: timeout,
-          enableParallelPush:
-            _.toLower(/** @type {'parallel'} */ (strategy)) === APP_INSTALL_STRATEGY.PARALLEL,
-        });
+        await B.resolve(pushFolder(afcService, app, bundlePathOnPhone, {enableParallelPush})).timeout(timeout);
         await this.installOrUpgradeApplication(
           bundlePathOnPhone,
-          await this.isAppInstalled(bundleId),
+          {
+            timeout: Math.max(timeout - timer.getDuration().asMilliSeconds, 60000),
```

Comment:
does this `60000` want to be `DEFAULT_APP_INSTALLATION_TIMEOUT_MS`? just to check the intention)

Reply:
that's ok. this is needed to make sure we at least allow one minute for app installation if files transfer has already almost used the limit